src/event-bridge/rule.ts

```diff
@@ -104,7 +104,7 @@ export interface IRule<T extends Event> {
    * Unsupported by Functionless:
    * * Variables from outside of the function scope
    */
-  map<P>(transform: EventTransformFunction<T, P>): EventTransform<T, P>;
+  map<E extends T, P>(transform: EventTransformFunction<E, P>): EventTransform<E, P>;
```

Comment:
This was the hard part to figure out. If you just use `T`, you get contra vs co-variant errors. This doesn't seem 100% right but it makes it all compile and infer.